### PR TITLE
chore: scheduled created-by migrations metrics

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

the created_by_user_id data migration from resolving events.created_by (for both events and features) now emits events on how many rows were updated.

Adds listeners for these events that records these metrics with prometheus

image

### CODE REVIEW

File: src/lib/features/events/event-store.ts [hunk 1]
```diff
@@ -433,16 +433,16 @@ class EventStore implements IEventStore {
         events.forEach((e) => this.eventEmitter.emit(e.type, e));
     }
 
-    async setCreatedByUserId(batchSize: number): Promise<void> {
+    async setCreatedByUserId(batchSize: number): Promise<number> {
         const API_TOKEN_TABLE = 'api_tokens';
 
         if (!this.flagResolver.isEnabled('createdByUserIdDataMigration')) {
-            return;
+            return -1;
```
Comment (gastonfournier) [inline src/lib/features/events/event-store.ts:440]: Find it weird that we return -1 here while we return Promise.resolve() down below... maybe this function should always return the number of rows updated and zero is meaningful, but -1 is not so much.

File: src/lib/features/feature-toggle/feature-toggle-service.ts [hunk 2]
```diff
@@ -2444,7 +2450,12 @@ class FeatureToggleService {
     }
 
     async setFeatureCreatedByUserIdFromEvents(): Promise<void> {
-        await this.featureToggleStore.setCreatedByUserId(100);
+        const updated = await this.featureToggleStore.setCreatedByUserId(100);
+        if (updated > -1) {
+            this.eventBus.emit(FEATURES_CREATED_BY_PROCESSED, {
+                updated,
+            });
+        }
```
Comment (gastonfournier) [inline src/lib/features/feature-toggle/feature-toggle-service.ts:2458]: I think it makes sense if we say that zero were updated to do this:
```suggestion
        if (updated > 0) {
            this.eventBus.emit(FEATURES_CREATED_BY_PROCESSED, {
                updated,
            });
        }
```

File: src/lib/features/feature-toggle/feature-toggle-store.ts [hunk 3]
```diff
@@ -723,13 +723,13 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
...
 
-    async setCreatedByUserId(batchSize: number): Promise<void> {
+    async setCreatedByUserId(batchSize: number): Promise<number> {
         const EVENTS_TABLE = 'events';
         const USERS_TABLE = 'users';
         const API_TOKEN_TABLE = 'api_tokens';
 
         if (!this.flagResolver.isEnabled('createdByUserIdDataMigration')) {
-            return;
+            return -1;
```
Comment (gastonfournier) [inline src/lib/features/feature-toggle/feature-toggle-store.ts:732]: Same suggestion
```suggestion
            return 0;
```

Comment (daveleek) [reply src/lib/features/feature-toggle/feature-toggle-store.ts:732]: I'll return undefined here instead as we discussed

File: src/lib/features/feature-toggle/feature-toggle-service.ts (hunk 2 above)
Comment (daveleek) [reply src/lib/features/feature-toggle/feature-toggle-service.ts:2458]: I'll change this to check for undefined

**Review by gastonfournier — APPROVED** [review]
LGTM!

**Pull request merged**